docs/designs/high-level.md

```diff
@@ -60,30 +61,34 @@ See [Custom Resource Reference Guide](crd.md) for a full reference to the custom
 
 The Reconcilers and Components communicate via a `ReconcilerContext` that is passed between them which they can use to share information during one reconcile run (for example the TLS reconciler uses it to tell the cluster reconciler to mount secrets containing TLS certificates to the pods). To communicate between runs and persist state the Status object of the Custom Resource can be used. It contains a data structure that exposes the current status of each reconciler and component. This has the double advantage of sharing state between Reconcilers and also make it visible to the user.
 
-The Manager retrieves the current custom resource, based on the ID, from Kubernetes (using its API) and checks to see whether it was deleted. If it was, nothing needs to be done since all Kubernetes objects created by the operator are using [Owner Reference](https://kubernetes.io/blog/2021/05/14/using-finalizers-to-control-deletion/#owner-references) to the custom resource, thus are deleted in a cascading fashion.
+The OpenSearch Reconciler retrieves the current custom resource, based on the ID, from Kubernetes (using its API) and checks to see whether it was deleted (by checking if the custom resource has deletion timestamp field in it). If it was, it removes the Finalizer that was placed on it, and calls `DeleteResources` for each Reconciler. All Kubernetes objects created by the operator are using [Owner Reference](https://kubernetes.io/blog/2021/05/14/using-finalizers-to-control-deletion/#owner-references) to the custom resource, thus are deleted in a cascading fashion by Kubernetes once the finalizer has been removed. 
+
+If it was created or updated, the OpenSearch Reconciler uses the retrieved custom resource YAML to parse it and produces a data structure representing it. It then executes the chain of Reconcilers, passing the Custom Resource data structure as main input.
+
+Each Reconciler executes and returns a special Result structure, that says two things:
+* Requeue: true/false. If true, it will fill the queue the controller is listening on with a message with the same resource ID. False - it won't do it, hence the OpenSearch Reconciler will only get called upon another change to the custom resource
+* RequeueAfter: units is seconds. If Requeue is true, the message will be inserted into the queue in RequeueAfter seconds.
```

Comment:
Technically the type is `time.Duration` so we can use whatever unit we want (e.g. `30 * time.Second`)